test/python/circuit/library/test_mcmt.py

```diff
@@ -290,7 +290,9 @@ def test_gate_with_parameters_vchain(self):
 
         self.assertEqual(circuit.count_ops().get("cry", 0), num_target)
         self.assertEqual(circuit.num_parameters, 1)
-        self.assertIs(circuit.parameters[0], theta)
+        # use equal since Rust's Parameter object can not compare by pointer
```

Comment:
Why can't the objects not be compared by pointer? (It's probably some basic PyO3/Rust thing I'm missing 🙂)

This should be documented very clearly, if we change the behavior.